### ISSUE TITLE
Keyboard is dismissed when tap on Return Key button

### ISSUE THREAD

**Issue by cuong1112035** [ReactiveX/RxSwift#1685]
I have some code like this: 

```
let search = textField.rx.text
            .filter { ($0 ?? "").count > 0 }
            .flatMapLatest { text in
                return ApiController.shared.currentWeather(city: text ?? "Error")
                    .catchErrorJustReturn(ApiController.Weather.dummy)
            }
            .observeOn(MainScheduler.instance)

        search.map { "\($0.temperature)° C" }
            .bind(to: tempLabel.rx.text)
            .disposed(by: bag)
```
The keyboard is dismissed every time I tap on Return Key button and I think this not the ios behavior. Every things work just well when I remove .bind(to:) operator.
Seem this issue is linkd to #1228 

**Platform/Environment**

  - [ x] iOS
  - [ ] macOS
  - [ ] tvOS
  - [ ] watchOS
  - [ ] playgrounds

**How easy is to reproduce? (chances of successful reproduce after running the self contained code)**

  - [ x] easy, 100% repro
  - [ ] sometimes, 10%-100%
  - [ ] hard, 2% - 10%
  - [ ] extremely hard, %0 - 2%

**Xcode version**:

```
Version 9.4 (9F1027a)
```

:warning: Fields below are optional for general issues or in case those questions aren't related to your issue, but filling them out will increase the chances of getting your issue resolved. :warning:

**Installation method**:
  - [x ] CocoaPods
  - [ ] Carthage
  - [ ] Git submodules

**I have multiple versions of Xcode installed**:
  (so we can know if this is a potential cause of your issue)
  - [ ] yes (which ones)
  - [x ] no

**Level of RxSwift knowledge**:
  (this is so we can understand your level of knowledge
    and formulate the response in an appropriate manner)
  - [ ] just starting
  - [x ] I have a small code base
  - [ ] I have a significant code base

**Comment by kzaher** [comment]
Hi @cuong1112035 ,

I believe this diff proves that this is Apple issue and not RxSwift issue

```diff
diff --git a/RxExample/RxExample/Examples/APIWrappers/APIWrappersViewController.swift b/RxExample/RxExample/Examples/APIWrappers/APIWrappersViewController.swift
index dceace60..a06f8c99 100644
--- a/RxExample/RxExample/Examples/APIWrappers/APIWrappersViewController.swift
+++ b/RxExample/RxExample/Examples/APIWrappers/APIWrappersViewController.swift
@@ -55,6 +55,11 @@ class APIWrappersViewController: ViewController {
 
     let manager = CLLocationManager()
 
+    @objc
+    func wtfAppleWTFFFFFF() {
+        print("WTF")
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
@@ -144,7 +149,10 @@ class APIWrappersViewController: ViewController {
         if #available(iOS 11.2, *) {
             // also test two way binding
             let textValue = Variable("")
-            _ = textField.rx.textInput <-> textValue
+            // _ = textField.rx.textInput <-> textValue
+
+            // comment and uncomment this line, this is not Rx API, but Rx uses this API internally
+            // textField?.addTarget(self, action: #selector(APIWrappersViewController.wtfAppleWTFFFFFF), for: .allEvents)
 
             textValue.asObservable()
                 .subscribe(onNext: { [weak self] x in
```

**Comment by cuong1112035** [comment]
Thanks so much, please help me to close this issue.
